### ISSUE TITLE
RLC-820A Pet motion sensor not working

### ISSUE THREAD

**Issue by craggyh** [fwestenberg/reolink_dev#512]
The "Pet" binary motion sensor doesn't work with 820A cameras on latest firmware. 
I get IOS notifications from the Reolink mobile app when a pet is detected but the motion sensor in HA never fires.

**Comment by GewoonRoy** [comment]
@craigueh What type of camera are you using, what firmware does your camera have and what does your json look like when you open the url:

http://`{your_camera_ip}`/api.cgi?cmd=getAiState&user=`{your_username}`&password=`{your_password}`

**Comment by craigueh** [comment]
Here's the relevan camera info:

- Model RLC-820A
- Build No.build 22041501
- Hardware No. IPC_523128M8MP
- Config Versionv3.1.0.0
- Firmware Versionv3.1.0.956_22041501
- Details IPC_523128M8MPS18E1W01100000001

and here's the json output:

[ { "cmd" : "GetAiState", "code" : 0, "value" : { "channel" : 0, "dog_cat"
: { "alarm_state" : 0, "support" : 1 }, "face" : { "alarm_state" : 0,
"support" : 0 }, "people" : { "alarm_state" : 0, "support" : 1 }, "vehicle"
: { "alarm_state" : 0, "support" : 1 } } } ]

**Comment by GewoonRoy** [comment]
Okay, well your json seems correct. I'm assuming that the sensor doesn't listen to "dog_cat" and there needs to be an update. But i might be wrong. Someone with more knowledge of python or the scripts of reolink_dev might help more.

**Comment by craigueh** [comment]
I managed to fix this by making the following changes to base.py and binary_sensor.py

base.py
line 134. Add the following:
self.object_pet_detection_state = True

binary_sensor.py
line 33. Change to:
base.sensor_pet_detection = ObjectDetectedSensor(hass, config_entry, "dog_cat")

line 218. Change to:
 if self._object_type == "dog_cat":

Restart Homeassistant and you should now have working pet detection sensors.

**Comment by skynet01** [comment]
Would be awesome to include this in the next release :)

**Comment by stalakerob** [comment]
Works for me too. Be aware that the sensor name changes to binary_sensor.(your camera name)_dog_cat_detected

**Comment by craigueh** [comment]
Can we get this fix added to the official release? It's a real nuisance having to edit the files to apply the fix every time the component is updated.

**Comment by 1danielsan** [comment]
Yes, would be fantastic if been added to the next official release... 
Thank you very much

**Comment by techolutions** [comment]
#579

**Comment by skynet01** [comment]
JimStar's version fixed this issue for me, along with other things  https://github.com/JimStar/reolink_cctv
